src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/MSBuildItem.cs

```diff
@@ -38,13 +38,33 @@ public MSBuildItem(string identity, IDictionary<string, string> metadata)
         }
 
         /// <summary>
+        /// Get property or null if empty. Trims whitespace from values.
+        /// </summary>
+        public string GetProperty(string property)
+        {
+            return GetProperty(property, trim: true);
+        }
+
+        /// <summary>
         /// Get property or null if empty.
         /// </summary>
-        public string GetProperty(string key)
+        public string GetProperty(string property, bool trim)
         {
             string val;
-            if (_metadata.TryGetValue(key, out val) && !string.IsNullOrEmpty(val))
+            if (_metadata.TryGetValue(property, out val) && val != null)
             {
+                if (trim)
+                {
+                    // Remove whitespace that occurs due to msbuild formatting.
+                    val = val.Trim();
+                }
+
+                if (string.IsNullOrEmpty(val))
+                {
+                    // Normalize empty values to null for consistency.
+                    val = null;
```

Comment:
`MSBuildTaskItem.GetProperty(...)` should use the same logic.  Currently, it uses the [inverse](https://github.com/NuGet/NuGet.Client/pull/1011/files#diff-265f20b267f9c7fef1df096a870975c7R47).